hasher-matcher-actioner/hmalib/lambdas/api/api_auth.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import os
+import json
+import time
+import jwt
+import requests
+import base64
+from jwt.algorithms import RSAAlgorithm
+
+from hmalib.common.logging import get_logger
+
+ACCESS_TOKEN = os.environ["ACCESS_TOKEN"]
+USER_POOL_URL = os.environ["USER_POOL_URL"]
+CLIENT_ID = os.environ["CLIENT_ID"]
+
+keys_url = f"{USER_POOL_URL}/.well-known/jwks.json"
+
+response = requests.get(keys_url)
+key_list = json.loads(response.text).get("keys", [])
+keys = {key["kid"]: json.dumps(key) for key in key_list}
+
+
+def is_jwt(token: str) -> bool:
+    jwt_split = token.split(".")
+    try:
+        if len(jwt_split) != 3:
+            return False
+        header = json.loads(base64.b64decode(jwt_split[0] + "=="))
+        if header["alg"] != "RS256":
+            return False
+    except Exception as e:
+        print("INFO: Provisioning Data is not in JWT format\n {0}".format(e))
+        return False
+    return True
+
+
+def validate_jwt(token: str):
+    logger = get_logger()
+
+    try:
+        if not keys:
+            logger.error("No JWT public keys found. User auth will always fail.")
+
+        kid = jwt.get_unverified_header(token)["kid"]
+        key = keys[kid]
+
+        public_key = RSAAlgorithm.from_jwk(key)
```

Comment:
Compute and cache this once per key. I see that [`from_jwk`](https://github.com/jpadilla/pyjwt/blob/master/jwt/algorithms.py#L301) method does some crypto work which can be avoided.